pkg/deploy/deploy.go

```diff
@@ -207,26 +207,18 @@ func manageResource(ctx context.Context, cli client.Client, res *resource.Resour
 		if !k8serr.IsNotFound(err) {
 			return err
 		}
-		// Create resource if it doesn't exist and component enabled
-		if enabled {
-			return createResource(ctx, cli, res, owner)
-		}
-		// Skip if resource doesn't exist and component is disabled
-		return nil
 	}
 
-	// when resource is found
+	// Component is enabled
 	if enabled {
-		// Exception to not update kserve with managed annotation
-		// do not reconcile kserve resource with annotation "opendatahub.io/managed: false"
-		// TODO: remove this exception when we define managed annotation across odh
-		if found.GetAnnotations()[annotations.ManagedByODHOperator] == "false" && componentName == "kserve" {
-			return nil
-		}
-		return updateResource(ctx, cli, res, found, owner)
+		// Create resource if it doesn't exist OR reconcile component if exists
+		return CreateOrUpdateResource(ctx, cli, res, found, owner)
+	}
+	// Component id removed: delete resource if it exists or do nothing if not found
```

Comment:
this is not easy to understand as there is a `found` object but also a check on the `IsNotFound` error, so the _do nothing if not found_ is a little bit misleading. IMHO, the `getResource` function should honor the `ErrNotFound` and there should be no need to check if `found != nil`